framework/src/base/Moose.C

```diff
@@ -27,6 +27,8 @@
 const ExecFlagType EXEC_NONE = registerDefaultExecFlag("NONE");
 const ExecFlagType EXEC_INITIAL = registerDefaultExecFlag("INITIAL");
 const ExecFlagType EXEC_LINEAR = registerDefaultExecFlag("LINEAR");
+const ExecFlagType EXEC_PRE_NONLINEAR_CONVERGENCE =
+    registerDefaultExecFlag("PRE_NONLINEAR_CONVERGENCE");
```

Comment:
What is the difference between `registerDefaultExecFlag` and `registerExecFlag`? This list is getting long, maybe it is a good idea to register this execute flag in `FEProblemSolve` because it is the solve object interfacing PETSc and will be using this flag.

Reply:
It controls what you get with `getDefaultFlags()` vs. `getFlags()`. `SetupInterface` has its `execute_on` use the defaults. If you want more in derived classes, you do this for example:
```
  ExecFlagEnum & exec_enum = params.set<ExecFlagEnum>("execute_on", true);
  exec_enum.addAvailableFlags(EXEC_PRE_MULTIAPP_SETUP);
```
You'd just need to do this for all of the types of object that need the new enum. I'm willing to do that here, just don't know what the preference is. Systems that will definitely need it include UserObjects (PPs) and whatever dependencies they end up having.

I don't know what you mean by the suggestion to register it to `FEProblemSolve`, since that one doesn't have an `execute_on`.

Reply:
Thanks.

I mean make this new `EXEC_PRE_NONLINEAR_CONVERGENCE` as a static member variable of `FEProblemSolve`.